pkg/aggregator/time_sampler.go

```diff
@@ -133,6 +130,13 @@ func (s *TimeSampler) flushSeries(cutoffTime int64) metrics.Series {
 		delete(s.counterLastSampledByContext, context)
 	}
 
+	return s.dedupSerieBySerieSignature(rawSeries)
+}
+
+func (s *TimeSampler) dedupSerieBySerieSignature(rawSeries []*metrics.Serie) []*metrics.Serie {
```

Comment:
I think it is quite obvious what the code does so it may be not require to have a comment explaining what the code does. It is unclear why the series have to be deduplicated and what is the meaning of `NameSuffix` and this part should be commented. I think it is because of distribution metrics which create several metrics, but other member of the team can provide a more detailed information.